### PR TITLE
Refactor keycloak integration

### PR DESCRIPTION
**Pull request by kingsleyzissou**
##### Checklist

- [ ] `npm test` passes
- [ ] `npm run build` works
- [ ] tests are included
- [ ] documentation is changed or added

### CODE REVIEW

File: client/src/AppContainer.tsx [hunk 1]
```diff
@@ -1,15 +1,14 @@
...
 import { AppContext } from './AppContext';
-import { keycloakConfig, clientConfig } from './config';
-import { keycloakEnabled, onKeycloakTokens } from './helpers/keycloakHelpers';
+import { clientConfig } from './config';
 import { Loading } from './components/Loading';
 import { IContainerProps } from './declarations';
+import { getKeycloakInstance } from './auth/keycloakAuth';
 
+let keycloak: any;
 const client = new ApolloOfflineClient(clientConfig);
```
Comment (wtrocki) [inline client/src/AppContainer.tsx]: In this context naming this graphqlClient or apolloClient will be much better. There is keycloak and other things happening here

File: client/src/auth/keycloakHelpers.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+// get tokens from localstorage
+export const getLocalTokens = () => {
+  const token = localStorage.getItem('token');
+  const refreshToken = localStorage.getItem('refreshToken');
+  return [token, refreshToken];
+};
+
+// save tokens to localstorage
+export const setLocalTokens = ({ token, refreshToken } : { token: string | undefined, refreshToken: string | undefined}) => {
```
Comment (wtrocki) [inline client/src/auth/keycloakHelpers.ts]: Out of curiosity - why we doing this? This is security risk as localstorage is not protected by domain same as cookies.

Comment (wtrocki) [reply client/src/auth/keycloakHelpers.ts]: Keycloak keeps tokens for this very reason in cookies.

Comment (wtrocki) [reply client/src/auth/keycloakHelpers.ts]: Keycloak on init should rely on cookie anyway

File: client/src/auth/keycloakHelpers.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    'onLoad': 'login-required',
+  };
+  return {
+    'promiseType': 'native',
+    'onLoad': 'login-required',
+  }
+}
+
+// retrieve token and get authorization header
+export const getAuthHeader = () => {
```
Comment (wtrocki) [inline client/src/auth/keycloakHelpers.ts]: Also this seems to be antipattern. Keycloak will provide way to refresh token.
See: https://github.com/aerogear/aerogear-js-sdk/blob/master/packages/auth/src/Auth.ts#L121

File: client/src/components/Router.tsx [hunk 4]
```diff
@@ -1,28 +1,23 @@
-import React, { useContext } from 'react';
+import React from 'react';
 import { Redirect, Route } from 'react-router-dom';
 import { IonApp, IonRouterOutlet } from '@ionic/react';
 import { IonReactRouter } from '@ionic/react-router';
-import { useKeycloak } from '@react-keycloak/web';
+// import { useKeycloak } fro+m '@react-keycloak/web';
```
Comment (wtrocki) [inline client/src/components/Router.tsx]:
```suggestion
```

**Comment by kingsleyzissou** [comment]
@wtrocki pushed the changes

File: client/src/auth/KeycloakRoute.tsx [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import React, { useContext } from 'react'
+import { Route } from 'react-router-dom'
+import { AppContext } from '../AppContext';
+import { useNetworkStatus } from 'react-offix-hooks';
+
+export const KeycloakRoute: React.FC<any> = ({ component: Component, ...rest }) => {
+  const isOnline = useNetworkStatus();
+  const { keycloak } = useContext(AppContext);
+
+  if (keycloak && !keycloak.authenticated && isOnline) keycloak?.login();
```
Comment (wtrocki) [inline client/src/auth/KeycloakRoute.tsx]:
```suggestion
  if (keycloak && !keycloak.authenticated && isOnline) keycloak.login();
```

File: client/src/auth/keycloakAuth.ts [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+      });
+    }
+  } catch {
+    keycloak = undefined;
+    console.error('Keycloak error: Unable to initialize keycloak');
+  }
+}
+
+
+// This function keeps getting called by wslink
```
Comment (wtrocki) [inline client/src/auth/keycloakAuth.ts]: /** are much better as they are part of the api. // are not

File: client/README.adoc [hunk 7]
```diff
@@ -0,0 +1,49 @@
+= React Client
+
+Example application using https://github.com/aerogear/offix[Offix Client] to provide additional offline capabilities on top of Apollo GraphQL.
```
Comment (wtrocki) [inline client/README.adoc]:
```suggestion
Example  React Web application using AeroGear DataSync GraphQL capabilities
```

**Pull request merged**